crates/core/thin_edge_json/src/event.rs

```diff
@@ -62,6 +71,26 @@ impl ThinEdgeEvent {
             ))
         }
     }
+
+    pub fn apply_defaults(&mut self) {
+        match &mut self.data {
+            None => {
+                self.data = Some(ThinEdgeEventData {
+                    text: Some(self.name.clone()),
+                    time: Some(OffsetDateTime::now_utc()),
+                    extras: HashMap::new(),
+                })
+            }
+            Some(event_data) => {
+                if event_data.text.is_none() {
+                    event_data.text = Some(self.name.clone());
+                }
+                if event_data.time.is_none() {
+                    event_data.time = Some(OffsetDateTime::now_utc());
+                }
+            }
+        }
+    }
```

Comment:
Wouldn't implementing `Default` trait be the way to go here?

Reply:
It couldn't be, because the "default" `text` field is derived from its own `name` field and not a static value. But, it doesn't matter because this function is not used anymore. I'll remove it.